tests/integ/modin/frame/test_loc.py

```diff
@@ -4068,6 +4068,74 @@ def test_df_loc_get_with_timedelta_and_none_key():
     assert_frame_equal(snow_df.loc[None], expected_df, check_column_type=False)
 
 
+@sql_count_checker(query_count=2, join_count=4)
+@pytest.mark.parametrize("index", [list("ABC"), [0, 1, 2]])
+def test_df_loc_set_row_from_series(index):
+    native_df = native_pd.DataFrame([[1, 2, 3], [4, 5, 6]], columns=list("ABC"))
+    snow_df = pd.DataFrame(native_df)
+
+    def locset(df):
+        series = (
+            pd.Series([1, 4, 9], index=index)
+            if isinstance(df, pd.DataFrame)
+            else native_pd.Series([1, 4, 9], index=index)
+        )
+        df.loc[1] = series
+        return df
+
+    eval_snowpark_pandas_result(
+        snow_df,
+        native_df,
+        locset,
+    )
+
+
+@sql_count_checker(query_count=2, join_count=1)
+@pytest.mark.parametrize("index", [[3, 4, 5], [0, 1, 2]])
+def test_df_loc_full_set_row_from_series_pandas_errors(index):
+    native_df = native_pd.DataFrame([[1, 2, 3], [4, 5, 6]])
+    snow_df = pd.DataFrame(native_df)
+
+    with pytest.raises(ValueError, match="setting an array element with a sequence."):
+        native_df.loc[:] = native_pd.Series([1, 4, 9], index=index)
+
+    def locset(df):
+        series = (
+            pd.Series([1, 4, 9], index=index)
+            if isinstance(df, pd.DataFrame)
+            else native_pd.Series([1, 4, 9], index=index)
+        )
+        if isinstance(df, pd.DataFrame):
+            df.loc[:] = series
+        else:
+            if index == [0, 1, 2]:
```

Comment:
you'd better just compare the result with the expected result. These steps here can be confusing.